src/main/java/org/sonatype/repository/conan/internal/metadata/ConanAbsoluteUrlIndexer.java

```diff
@@ -0,0 +1,203 @@
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.sonatype.goodies.common.ComponentSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.nexus.repository.storage.TempBlob;
+import org.sonatype.nexus.repository.transaction.TransactionalStoreMetadata;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Supplier;
+
+import static java.nio.charset.Charset.defaultCharset;
+import static java.util.Collections.emptyMap;
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+import static org.sonatype.nexus.repository.view.Content.maintainLastModified;
+import static org.sonatype.nexus.repository.view.ContentTypes.APPLICATION_JSON;
+import static org.sonatype.repository.conan.internal.AssetKind.CONAN_SRC;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.HASH_ALGORITHMS;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.findAsset;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.toContent;
+
+/**
+ * download_url files contain absolute paths to each asset
+ *
+ * This class removes the absolute address so as to redirect back to this repository
+ *
+ * @since 3.conan
+ */
+@Singleton
+@Named
+public class ConanAbsoluteUrlIndexer
+    extends ComponentSupport
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Nullable
+  public TempBlob updateAbsoluteUrls(final TempBlob tempBlob,
+                                     final Repository repository,
+                                     final String assetName) {
+    Map<String, URL> currentURLMap = readIndex(tempBlob.get(), assetName);
+    Map<String, URL> indexes = new HashMap<>();
+
+    for (Map.Entry<String, URL> entry : currentURLMap.entrySet()) {
+      URL originalUrl = entry.getValue();
+      URL indexUrl = getIndexedUrl(repository.getUrl(), entry.getValue().getPath());
+      indexes.put(entry.getValue().getPath(), originalUrl);
+      entry.setValue(indexUrl);
+    }
+
+    handleUpdatingIndexes(assetName, indexes, repository);
+
+    try {
+      return convertFileToTempBlob(MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(currentURLMap), repository);
+    }
+    catch (JsonProcessingException e) {
+      log.warn("Unable to write to blob", e);
+      return null;
+    }
+  }
+
+  private URL getIndexedUrl(final String localHost, final String path) {
```

Comment:
Minor: This isn't strictly "localhost"; if the user is accessing Nexus remotely (which is most common) then this will either be a domain name or an IP address. I wonder if "repositoryUrl" would be less misleading? Also have you double checked that it works as expected for non localhost urls? I've had a look through the BaseUrlHolder code and I'd expect it to work (we do something similar for docker token auth) but I'd recommend a quick test.